generate_makefile.bash

```diff
@@ -355,6 +359,14 @@ if [[ ${KOKKOS_DEVICE_CMD} == *Kokkos_ENABLE_HPX* ]]; then
    fi
 fi
 
+## if SYCL is enabled, we need to enforce cxx standard = 17
+if [[ ${KOKKOS_DEVICE_CMD} == *Kokkos_ENABLE_SYCL* ]]; then
+   if [ "${KOKKOS_CXX_STANDARD}" == "" ] || [ ${#KOKKOS_CXX_STANDARD} -lt 17 ]; then
+      echo CXX Standard must be 17 or higher for SYCL to work.
+      KOKKOS_CXX_STANDARD=17
+   fi
+fi
```

Comment:
We would need to require C++17 in CMake as well.
@nliber can you remind me why we can't use C++14?

Reply:
So far I have not seen that I need to explicitly have to provide a standard flag.

Reply:
SYCL 2020 is requiring C++17.